dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/intermediate/components/basecomponents/NumberBaseComponent.kt

```diff
@@ -0,0 +1,49 @@
+package org.dataland.frameworktoolbox.intermediate.components.basecomponents
+
+import org.apache.commons.text.StringEscapeUtils
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.intermediate.components.ComponentBase
+import org.dataland.frameworktoolbox.intermediate.components.addStandardCellWithValueGetterFactory
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+
+/**
+ * The NumberBaseComponent serves as base-class for any number component (percentage, decimal, integer...) and
+ * offers some standard functionality.
+ */
+open class NumberBaseComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(identifier, parent) {
+    open var constantUnitSuffix: String? = null
+
+    override fun generateDefaultDataModel(dataClassBuilder: DataClassBuilder) {
+        dataClassBuilder.addProperty(
+            this.identifier,
+            documentSupport.getJvmTypeReference(
+                TypeReference("java.math.BigDecimal", isNullable),
```

Comment:
If the constructor had a parameter `fullyQualifiedName` and pass it to `TypeReference` instead of `java.math.BigDecimal` Then in no subclass be the need for overwriting this method because the subclasses could control it themselves.

Reply:
I will investigate this idea and come back to this conversation afterwards.

Reply:
I have pushed a change for this.   Can you check it and give me feedback ?

Reply:
Love it.